apps/arweave/src/ar_tx.erl

```diff
@@ -236,12 +246,14 @@ signature_data_segment_v1(T) ->
 		(tags_to_binary(T#tx.tags))/binary
 	>>.
 
-sign(TX, PrivKey, PubKey, SignatureDataSegment) ->
-	NewTX = TX#tx{ owner = PubKey },
+sign(TX, PrivKey, {_, Owner}, SignatureDataSegment) ->
+	NewTX = TX#tx{ owner = Owner },
 	Sig = ar_wallet:sign(PrivKey, SignatureDataSegment),
 	ID = crypto:hash(?HASH_ALG, <<Sig/binary>>),
 	NewTX#tx {
-		signature = Sig, id = ID
+		id = ID,
+		signature = Sig
+		% signature_type = KeyType
```

Comment:
We need to uncomment this line to allow node operators to send transactions signed with ECDSA or EdDSA via the `/unsigned_tx` endpoint but that would require that we introduce a height parameter to this function.

What if we just enabled this after the fork?

Reply:
Yeah, I do not think (m)any people are using this endpoint. Anyway, it won't be difficult to add support for other signatures later at any moment.